src/services/gas-price/providers/arbitrum.provider.ts

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+import { GasPrice } from './types';
+import { bnum } from '@/lib/utils';
+
+interface ArbitrumGasStationResponse {
+  id: number;
+  jsonrpc: string;
+  result: string;
+}
+
+export default class ArbitrumProvider {
+  public async getGasPrice(): Promise<GasPrice | null> {
+    try {
+      const [gasPrice, maxPriorityFee] = await Promise.all([
+        this.fetchArbitrumProvider('eth_gasPrice'),
+        this.fetchArbitrumProvider('eth_maxPriorityFeePerGas'),
+      ]);
+      const price = bnum(gasPrice.result).toNumber();
+      const maxPriorityFeePerGas = bnum(maxPriorityFee.result).toNumber();
+
+      return {
+        price,
+        maxPriorityFeePerGas,
+      };
+    } catch (error) {
+      console.log('[Arbitrum] Gas Platform Error', error);
+      return null;
+    }
+  }
+
+  private async fetchArbitrumProvider(method: string) {
+    const { data } = await axios.post<ArbitrumGasStationResponse>(
+      // TO-DO
+      // if use configService.loggingRpc - test fails as jest somehow uses goerli config
+      'https://arb-mainnet.g.alchemy.com/v2/VBeQgTCRqqPtuuEPsFzRdwKXzDyN6aFh',
```

Comment:
We could equally be using infura here? I don't think it's a good idea to hardcode this URL. I know we have for polygon but that's because it's a specific API for fetching the gas price. Here you are using an RPC, please inject configService as a dependency and use the `getNetworkRpc` function to fetch the URL: https://github.com/balancer-labs/frontend-v2/blob/develop/src/services/config/config.service.ts#L62-L69

Reply:
yes, it worked. thanks @garethfuller !